### DISCUSSION TITLE
PDFs' Download Ability and Work with Them

### DISCUSSION THREAD

**Discussion by PavelMorozovskiy** [n3d1117/chatgpt-telegram-bot#484] [Ideas]
The ability to upload pdf files to the bot and work with them. I saw this approach in working with the creation of a Streamlit application (the example of 'Chat with PDFs' app- https://youtu.be/dXxQ0LR-3Hg?si=QgWNFTFlvrI9B4k0)

**Comment by alexk-epsilon** [comment]
It seems that chat.completions API does not support pdf files at the moment, which leaves us with two options:
1) Use message or assistant api. This will let the bot respond to simple queries about the document, similar to the query in the video (wht are the 3 branches of the us govt ?). But since the rest of the bot uses chat.completions, this will lack context and effectively ignore the rest of the conversation (because the rest of the conversation is attached to one of the chat completions). 
2) Use a third party library to convert pdf to text and then send it to openai as a regular text message. This will probably limit the size of the document that can be passed to openai in that manner. Can also try splitting larger files into chunks and send one by one. It will probably be a not-so-budget friendly option as well if the file is large. 
No matter which option is selected, it will likely have to be refactored once chat.completions starts supporting files.
The above applies to other "text-like" document formats (docx, ps, rtf, etc.)
Which option makes more sense to you guys ? 🙃

**Reply by alexk-epsilon** [reply]
Here's yet another option. Convert pdf to image and feed it to vision model 🙃
